spec/controllers/course/levels_controller_spec.rb

```diff
@@ -6,42 +6,37 @@
 
   with_tenant(:instance) do
     let!(:user) { create(:administrator) }
+    let!(:normal_user) { create(:user) }
     let!(:course) { create(:course) }
-    let!(:level_stub) do
-      stub = create(:course_level, course: course)
-      allow(stub).to receive(:save).and_return(false)
-      allow(stub).to receive(:destroy).and_return(false)
-      stub
-    end
-
-    before { sign_in(user) }
 
-    describe '#destroy' do
-      subject { delete :destroy, params: { course_id: course, id: level_stub } }
+    describe '#create' do
+      context 'when user is allowed to edit levels' do
+        before { sign_in(user) }
 
-      context 'when destroy fails' do
-        before do
-          controller.instance_variable_set(:@level, level_stub)
-          subject
-        end
+        it 'is expected to create new levels' do
+          post :create, params: { course_id: course.id, levels: [0, 100, 200, 400] }, format: :json
+          saved_levels = course.reload.levels.map(&:experience_points_threshold)
 
-        it { is_expected.to redirect_to(course_levels_path(course)) }
-        it 'sets an error flash message' do
-          expect(flash[:danger]).to eq(I18n.t('course.levels.destroy.failure', error: ''))
+          expect(saved_levels).to match_array([0, 100, 200, 400])
```

Comment:
You can leave this test here, but I would prefer tests which test `mass_update_levels`.

Reply:
Added model tests.